### PR TITLE
Feat/auth

### PR DESCRIPTION
**Pull request by Debarshi95**
PR for auth pages SignIn and Signup with Firebase as backend

preview : https://624c00cb597fd17cb266070c--debarshib-flixnotes.netlify.app/signin

### CODE REVIEW

**Review by m26unkwn — COMMENTED** [review]
The code is very well written.

Everything is smooth and clean.

**Review by m26unkwn — APPROVED** [review]

File: src/components/common/Button/Button.css [hunk 1]
```diff
@@ -0,0 +1,32 @@
+.Button__root {
+  border: 0;
+  display: flex;
+  text-align: center;
```
Comment (muskanbagrecha) [inline src/components/common/Button/Button.css]: I don't think you require text-align center as flex is centering it already.

File: src/components/common/Button/Button.css [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+.Button--contained {
+  background-color: var(--color-primary);
+  color: var(--color-text-white);
+  padding: 0.4rem 1rem;
+}
+
+.Button--outlined {
+  border: 1.8px solid var(--color-gray);
```
Comment (muskanbagrecha) [inline src/components/common/Button/Button.css]: You can round this off.

File: src/components/common/Navbar/Navbar.css [hunk 3]
```diff
@@ -0,0 +1,7 @@
+.Navbar__root {
+  padding: 1.25rem 3rem;
+  border: 0.9px solid var(--color-main-bg);
```
Comment (muskanbagrecha) [inline src/components/common/Navbar/Navbar.css]: You can round this off.

File: src/components/common/Navbar/Navbar.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    <div className="Navbar__root">
+      <nav className="d-flex content-between max-w-85 mx-auto items-center">
+        <NavLink to="/" className="Navlink--primary text-14 text-bold">
+          FlixNote
+        </NavLink>
+        <div className="d-flex w-10 content-between items-center">
+          {user?.id ? (
+            <>
+              <NavLink to="/cart" className="Navbar__IconButton">
+                {/* <BsFillCartFill /> */}
```
Comment (muskanbagrecha) [inline src/components/common/Navbar/Navbar.js:18]: Remove commented code.

File: src/components/common/Typography/Typography.css [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+}
+.Typography.Typography--ssm {
+  font-size: 1.25rem;
+}
+
+.Typography.Typography--md {
+  font-size: 1.5rem;
+}
+.Typography.Typography--primary {
+  color: var(--color-primary) !important;
```
Comment (muskanbagrecha) [inline src/components/common/Typography/Typography.css:50]: Please avoid using important here. You can increase the specificity here.

File: src/pages/auth/Signin/Signin.css [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+}
+
+.Signin__formContainer {
+  max-width: 26rem;
+  margin: 0 auto;
+  padding: 2rem;
+  position: relative;
+  transform: translate(0, 10%);
+  border-radius: 4px;
+  box-shadow: rgba(0, 0, 0, 0.16) 0px 1px 4px;
```
Comment (muskanbagrecha) [inline src/pages/auth/Signin/Signin.css]: Please declare this as a variable for consistency.

File: src/pages/auth/Signin/Signin.js [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+      <div className="Signin__formContainer d-flex flex-col">
+        <Typography variant="h5" className="Typography--primary mt-2 mb-1 text-center">
+          Signin to continue
+        </Typography>
+        <Formik
+          initialValues={{
+            email: '',
+            password: '',
+          }}
+          validationSchema={validateLogin()}
```
Comment (muskanbagrecha) [inline src/pages/auth/Signin/Signin.js:51]: I think you should pass the function as a reference here.

File: src/pages/auth/Signin/Signin.js [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+                  />
+                  <Button
+                    component="button"
+                    type="submit"
+                    variant="contained"
+                    className="Signin__button text-bold w-10 text-10"
+                    disabled={
+                      isSubmitting ||
+                      !isValid ||
+                      Boolean(values.email === '' || values.password === '')
```
Comment (muskanbagrecha) [inline src/pages/auth/Signin/Signin.js:90]: Why do we have to type cast into Boolean?

File: src/pages/auth/Signup/Signup.css [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+  width: 100%;
+}
+
+.Signup__formContainer {
+  max-width: 26rem;
+  margin: 2rem auto;
+  padding: 1rem 2rem;
+  position: relative;
+  border-radius: 4px;
+  box-shadow: rgba(0, 0, 0, 0.16) 0px 1px 4px;
```
Comment (muskanbagrecha) [inline src/pages/auth/Signup/Signup.css]: Please use variables.

**Review by muskanbagrecha — APPROVED** [review]
Great work. Loved how you have abstracted the components.

File: src/pages/auth/Signin/Signin.js (hunk 7 above)
Comment (Debarshi95) [reply src/pages/auth/Signin/Signin.js:51]: Didn't get the context here.

File: src/components/common/Typography/Typography.css (hunk 5 above)
Comment (Debarshi95) [reply src/components/common/Typography/Typography.css:50]: This is intentional and added for a specific use-case

File: src/pages/auth/Signin/Signin.js (hunk 8 above)
Comment (Debarshi95) [reply src/pages/auth/Signin/Signin.js:90]: Not type-casting, Boolean call can be removed though. Thanks

**Pull request merged**